### PR TITLE
AGDIGGER-180

### PR DESCRIPTION
**Pull request by odra**
* Added support for 2+ mac nodes on jenkins
* Fixed the usage of mac node keys when specifying a key path

Note: I noticed some weird behavior on jenkins that it sometimes needed a reboot to show all added mac node

### CODE REVIEW

File: provision-osx/tasks/configure_buildfarm_node.yml [hunk 1]
```diff
@@ -3,42 +3,64 @@
...
+  block:
   -
     name: Generate SSH key pair for Jenkins
     command: ssh-keygen -t rsa -f $HOME/.ssh_jenkins/jenkins_id_rsa -N "{{ credential_passphrase | default('') }}"
     args:
-      creates: "$HOME/.ssh_jenkins/jenkins_id_rsa"
+      creates: "{{ node_key }}"
+    when: (credential_private_key is undefined) and (credential_public_key is undefined)
+  when:
+    - (credential_private_key is undefined) or (credential_public_key is undefined)
```
Comment (matskiv) [inline provision-osx/tasks/configure_buildfarm_node.yml]: If I understood this code correctly, this code generates key pair on each macOS node, but in jenkins we only have one instance of credentials. So whichever node uploads credential last, will work and others won't.

**Comment by matskiv** [comment]
I believe that the problem I described earlier is still present. We generate key on each MacOS node but upload it to jenkins under one Credential item so it gets overridden if we have multiple MacOS nodes.

**Comment by matskiv** [comment]
I changed some logic in @odra code. Key is no longer uploaded on each MacOS node, just master.
I tested my code by creating two MacOS slaves at once and also by creating them separately.

I noticed same jenkins behavior as described by odra in description. Only one slave is visible in jenkins after creation. I could see both only after reboot. I will try to add some clever logic to do a reboot when new slaves are created.

**Comment by matskiv** [comment]
There was a race condition because we used same temporary file for all nodes, so I randomized the name.
I also added handler to do a restart if new node was created, this way we work around the weird behavior that is in description. Handler will only run once and will wait for user input before doing the restart.

This ticket is ready for verification.

**Comment by matskiv** [comment]
@laurafitzgerald You could do some app build, but this changes are only related to node creation in jenkins so if you haven't run full installation, build might fail for some unrelated reason. I think it should be enough if build starts on one of the nodes.

**Comment by laurafitzgerald** [comment]
@matskiv triggered two concurrent app builds which were schedule one to each of the macs. it failed on npm install on one the machines but as you mentioned above I don't believe this is related to this change. Happy to approve this one.

**Pull request merged**